File: solutions/julia/saddle-points/saddle-points.jl
function saddlepoints(M)

    if M == []
        return []
    end

    rows, cols = size(M)
    candidates = NTuple{2,Int}[]

    for i in 1:rows
        row_max = maximum(M[i, :])
        for j in 1:cols
            column_min = minimum(M[:, j])
            if M[i, j] == row_max && M[i, j] == column_min
                push!(candidates, (i, j))
            end
        end
    end

    return candidates
end

saddlepoints([])

File: solutions/csharp/saddle-points/SaddlePoints.cs
public static class SaddlePoints
{
    public static IEnumerable<(int, int)> Calculate(int[,] matrix)
    {
        var candidates = new List<(int, int)>();
        for (int i = 0; i < matrix.GetLength(0); i++)
        {
            var row_max = Enumerable.Range(0, matrix.GetLength(1)).Max(c => matrix[i, c]);
            for (int j = 0; j < matrix.GetLength(1); j++)
            {
                var column_min = Enumerable.Range(0, matrix.GetLength(0)).Min(c => matrix[c, j]);
                if (matrix[i, j] == row_max && matrix[i, j] == column_min)
                {
                    candidates.Add((i + 1, j + 1));
                }
            }
        }
        return candidates;
    }
}
